### ISSUE TITLE
kvm test run keeps getting unresolved, yes it's getting worse

### ISSUE THREAD

**Issue by cagney** [libreswan/libreswan#1846]
This is one reason:
https://gitlab.com/libvirt/libvirt/-/issues/683

**Comment by cagney** [comment]
work-around had this crash:
```
ikev2-55-ipseckey-06 tb2 3:48:08.05: stop processing test ikev2-55-ipseckey-06 (test 986 of 1283) after 128.7 seconds
Exception in thread tb2:
Traceback (most recent call last):
  File "/usr/lib64/python3.12/threading.py", line 1075, in _bootstrap_inner
kvmrunner 3:48:09.02: waiting for threads to finish
    self.run()
  File "/usr/lib64/python3.12/threading.py", line 1012, in run
    self._target(*self._args, **self._kwargs)
  File "/home/libreswan/benchdir/testing/utils/fab/runner.py", line 575, in _process_test_queue
    _process_test(domain_prefix, domains, args,
  File "/home/libreswan/benchdir/testing/utils/fab/runner.py", line 510, in _process_test
    test_domain.stop()
  File "/home/libreswan/benchdir/testing/utils/fab/runner.py", line 115, in stop
    self.domain.destroy()
  File "/home/libreswan/benchdir/testing/utils/fab/virsh.py", line 134, in destroy
    console = self.console()
              ^^^^^^^^^^^^^^
  File "/home/libreswan/benchdir/testing/utils/fab/virsh.py", line 227, in console
    return self._open_console()
           ^^^^^^^^^^^^^^^^^^^^
  File "/home/libreswan/benchdir/testing/utils/fab/virsh.py", line 206, in _open_console
    if self._console.expect([pexpect.EOF,
       ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pexpect/spawnbase.py", line 354, in expect
    return self.expect_list(compiled_pattern_list,
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pexpect/spawnbase.py", line 383, in expect_list
    return exp.expect_loop(timeout)
           ^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pexpect/expect.py", line 181, in expect_loop
    return self.timeout(e)
           ^^^^^^^^^^^^^^^
  File "/usr/lib/python3.12/site-packages/pexpect/expect.py", line 144, in timeout
    raise exc
pexpect.exceptions.TIMEOUT: Timeout exceeded.
<fab.console.Console object at 0x7fe4161587d0>
command: /usr/bin/sudo
args: ['/usr/bin/sudo', 'virsh', '--connect', 'qemu:///system', 'console', '--force', 'tb2road']
buffer (last 100 chars): b''
before (last 100 chars): ''
after: <class 'pexpect.exceptions.TIMEOUT'>
match: None
match_index: None
exitstatus: None
flag_eof: False
pid: 719867
child_fd: 7
closed: False
timeout: 10
delimiter: <class 'pexpect.exceptions.EOF'>
logfile: None
logfile_read: <fab.console._Debug object at 0x7fe41615b8f0>
logfile_send: <fab.console._Debug object at 0x7fe41615b2c0>
maxread: 2000
ignorecase: False
searchwindowsize: None
delaybeforesend: 0.05
delayafterclose: 0.1
delayafterterminate: 0.1
searcher: searcher_re:
    0: EOF
    1: re.compile(b'Connected to domain tb2road\r\nEscape character is \\^]')
    2: re.compile(b"Connected to domain 'tb2road'\r\nEscape character is \\^] \\(Ctrl \\+ ]\\)\r\n")
ikev2-55-ipseckey-05 tb. 3:48:08.08: stop processing test ikev2-55-ipseckey-05 (test 985 of 1283) after 129.3 seconds
```
seems attempting to open a console can hang

**Comment by cagney** [comment]
this is libvirt's QEMU line, good starting point:
```
/usr/bin/qemu-system-x86_64 \
-name guest=m.road,debug-threads=on \
-S \
-object {"qom-type":"secret","id":"masterKey0","format":"raw","file":"/var/lib/libvirt/qemu/domain-28946-m.road/master-key.aes"} \
-machine pc-i440fx-8.2,usb=off,dump-guest-core=off,memory-backend=pc.ram,acpi=on \
-accel kvm \
-cpu qemu64 \
-m size=524288k \
-object {"qom-type":"memory-backend-ram","id":"pc.ram","size":536870912} \
-overcommit mem-lock=off \
-smp 1,sockets=1,cores=1,threads=1 \
-uuid 367598af-8e37-48d2-96c1-802005443b2a \
-display none -no-user-config \
-nodefaults \
-chardev socket,id=charmonitor,fd=29,server=on,wait=off \
-mon chardev=charmonitor,id=monitor,mode=control \
-rtc base=utc \
-no-shutdown \
-boot strict=on \
-kernel /home/libreswan/pool/m.linux-transmogrify.vmlinuz \
-initrd /home/libreswan/pool/m.linux-transmogrify.initramfs \
-append root=/dev/vda2 ro quiet console=tty0 plymouth.enable=0 mitigations=off biosdevname=0 net.ifnames=0 console=ttyS0,115200 \
-device {"driver":"piix3-usb-uhci","id":"usb","bus":"pci.0","addr":"0x1.0x2"} \
-device {"driver":"virtio-serial-pci","id":"virtio-serial0","bus":"pci.0","addr":"0x6"} \
-fsdev local,security_model=none,id=fsdev-fs0,path=/home/libreswan/git-misc/testing \
-device {"driver":"virtio-9p-pci","id":"fs0","fsdev":"fsdev-fs0","mount_tag":"testing","bus":"pci.0","addr":"0x2"} \
-fsdev local,security_model=none,id=fsdev-fs1,path=/home/libreswan/git-misc \
-device {"driver":"virtio-9p-pci","id":"fs1","fsdev":"fsdev-fs1","mount_tag":"source","bus":"pci.0","addr":"0x3"} \
-fsdev local,security_model=none,id=fsdev-fs2,path=/home/libreswan/pool \
-device {"driver":"virtio-9p-pci","id":"fs2","fsdev":"fsdev-fs2","mount_tag":"pool","bus":"pci.0","addr":"0x4"} \
-netdev {"type":"tap","fd":"31","vhost":true,"vhostfd":"38","id":"hostnet0"} \
-device {"driver":"virtio-net-pci","netdev":"hostnet0","id":"net0","mac":"12:00:00:ab:cd:02","bus":"pci.0","addr":"0x5"} \
-chardev pty,id=charserial0 \
-device {"driver":"isa-serial","chardev":"charserial0","id":"serial0","index":0} \
-audiodev {"id":"audio1","driver":"none"} \
-device {"driver":"virtio-balloon-pci","id":"balloon0","bus":"pci.0","addr":"0x9"} \
-object {"qom-type":"rng-random","id":"objrng0","filename":"/dev/random"} \
-device {"driver":"virtio-rng-pci","rng":"objrng0","id":"rng0","bus":"pci.0","addr":"0x7"} \
-sandbox on,obsolete=deny,elevateprivileges=deny,spawn=deny,resourcecontrol=deny \
-msg timestamp=on
```

**Comment by cagney** [comment]
this is another issue https://gitlab.com/libvirt/libvirt/-/issues/761

**Comment by cagney** [comment]
everything has a workaround
